crates/ruff_server/src/trace.rs

```diff
@@ -48,13 +52,26 @@ pub(crate) fn init_tracing(
         .set(sender)
         .expect("logging sender should only be initialized once");
 
-    let log_file = log_file.and_then(|path| {
-        std::fs::OpenOptions::new()
-            .create(true)
-            .append(true)
-            .open(path)
-            .ok()
-    });
+    let log_file = log_file
+        .map(|path| {
+            // this expands `logFile` so that tildes and environment variables
+            // are replaced with their values, if possible.
+            if let Some(expanded) = shellexpand::full(&path.to_string_lossy())
+                .ok()
+                .and_then(|path| PathBuf::from_str(&path).ok())
+            {
+                expanded
+            } else {
+                path.to_path_buf()
+            }
+        })
+        .and_then(|path| {
+            std::fs::OpenOptions::new()
+                .create(true)
+                .append(true)
+                .open(path)
+                .ok()
```

Comment:
Unrelated to this change: I think it would be good to write to stderr if opening the logfile fails instead of failing silently